sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -600,6 +600,7 @@ class SparkSession private(
    * @since 2.0.0
    */
   def sql(sqlText: String): DataFrame = {
+    SparkSession.setActiveSession(this)
```

Comment:
Who takes responsibility to set this back if we set the active session in this method? For example, the current active session is `spark1` when calling `spark2.sql(...)`, then the active session will become `spark2`. When we will set it back to `spark1`, or it will always be `spark2`?

Reply:
> Who takes responsibility to set this back if we set the active session in this method? For example, the current active session is `spark1` when calling `spark2.sql(...)`, then the active session will become `spark2`. When we will set it back to `spark1`, or it will always be `spark2`?

Under normal circumstances， our spark program use one SparkSession,  if some one write program use multi sparksession, they should control these things. 

And for analyzer, it will set this.

https://github.com/apache/spark/blob/e99a9f78ea2e18eabca722c5714280eb7a737499/sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala#L64

And only 
`SparkSession.cleanupAnyExistingSession()` will call 
```
  SparkSession.clearActiveSession()
      SparkSession.clearDefaultSession()
```